Define problem (#1)
travissanderson-wf/dev.dsmhack.org: content/project-framing.md

---
title: Framing the Project
home_spotlights:
  enabled: false
  weight: 1
  excerpt: The success of your project largely depends on understanding your nonprofit's project.
      Before you crack open your IDE and dive into the code, take some time to glean important insights
      up front, so that you can be more efficient with your time during the hack.
  home_img:
    path: images/framing-barn.jpg
    data_position: center center
    url: "#"
layout: page
---

Congratulations! You've selected your nonprofit project and are ready to get started. Before you crack open your IDE and dive into the code, the first few hours of the jam are most well spent building a deep understanding of the problem your nonprofit is trying to solve. By doing this well, you will have a better defined path for the next 40+ hours. You may also discover unexpected solutions that your nonprofit didn't even know to ask for. After all, they are experts in serving their communities, not the nuances of the latest and greatest technology.

## Interviewing Your Nonprofit

### Take (Post It) Notes

> A lot is about to be said in the next hour. You cannot possibly remember it all. You also don't know enough yet to understand how what they are saying this moment connects to an insight you'll have in two hours. You could do this in a notebook or shared document, like your probably used to for school, or you could try something a bit different.

Author: ```suggestion
A lot is about to be said in the next hour. You cannot possibly remember it all. You also don't know enough yet to understand how what they are saying this moment connects to an insight you'll have in two hours. Your first instinct may be to do this in a notebook or shared document, but the following "low tech" approach has a number of benefits.
```

Author: if not the whole rewording suggestion, there is a typo "like your probably" -> "like you're probably"

> PostIt Notes have become almost [synonymous with brainstorming & creativity](https://www.nngroup.com/articles/post-it-in-ux/) for good reason. They are flexible, tangible, collaborative, and low tech. By taking notes on PostIts, you can quickly connect and organize ideas in a way that both your team and nonprofits can quickly see and contribute to. Jot one idea per note. Come up with a loose color coding (like blue for notes, yellow for ideas, red for pain points, or you're fine just sticking to one color if it is too much to keep track of). Don't worry too much about organizing them at first. You can start by putting them randomly on a wall or even a table. Once the ideas are out there, you can start grouping them. Stick related notes to one another so you can move the group all together.

Author: ```suggestion
PostIt Notes have become almost [synonymous with brainstorming & creativity](https://www.nngroup.com/articles/post-it-in-ux/) for good reason. They are flexible, tangible, collaborative, and low tech (no account setup required; not even a computer!). By taking notes on PostIts, you can quickly connect and organize ideas in a way that both your team and nonprofits can quickly see and contribute to. Jot one idea per note. Come up with a loose color coding (like blue for notes, yellow for ideas, red for pain points, or you're fine just sticking to one color if it is too much to keep track of). Don't worry too much about organizing them at first. You can start by putting them randomly on a wall or even a table. Once the ideas are out there, you can start grouping them. Stick related notes to one another so you can move the group all together.
```

Author: ```suggestion
PostIt Notes have become almost [synonymous with brainstorming & creativity](https://www.nngroup.com/articles/post-it-in-ux/) for good reason. They are flexible, tangible, collaborative, and low tech. By taking notes on PostIts, you can quickly connect and organize ideas in a way that both your team and nonprofits can quickly see and contribute to. Jot one idea per note. Come up with a loose color coding (like blue for notes, yellow for ideas, red for pain points, or you're fine just sticking to one color if it is too much to keep track of). Don't worry about organizing them at first or people writing the same thing on different notes. You can start by putting them randomly on a wall or even a table. Once the ideas are out there, you can start grouping them. Stick related notes to one another so you can move the group all together.
```

### Active & Open Listening
The focus of this time is to hear what they have to say, not to try to push your own solution or ideas (wait for it, that can come later). By using [active listening](https://www.mindtools.com/CommSkll/ActiveListening.htm) and open interview questions, you'll get much richer information.
- **Pay attention.** This is not the time to be multi-tasking on your phone.
- **Repeat, reframe & summarize what you're hearing.** "So you're saying it would be helpful to... Did I get that right?"
- **Ask questions to clarify & get below the surface.** "What do you mean when you say..." "I heard you say... Tell me more about that."
- **Let them talk.** Try to avoid talking much beyond your questions. Don't interrupt and be ok with some silence after your questions.
- **Dig into the why.** Anytime they veer into proposing solutions, dig deeper. "You say you need an app. Why?" "Tell me more about how X would change things." The technique of ["5 Whys"](https://www.mindtools.com/pages/article/newTMC_5W.htm) can surface an underlying need that may have simpler solutions than what they realize.
- **Start broad, get more specific.**

> - **Ask for examples.** Get specific. "The last time" or "the worst time" is usually much more detailed than the general average version of the story.

Author: ```suggestion
- **Ask for examples of the problem they're trying to solve.** Get specific. "The last time" or "the worst time" is usually much more detailed than the general average version of the story.
```

- **What accounts & passwords to existing systems might be necessary?** If they are asking to integrate into a current technology system, they should either create an account for you to log in or give you the credentials to log in. You'd hate to be making great progress at 2am only to be blocked by an unknown password.
- **What's the best way to contact them during the jam?** Exchange cell numbers, email addresses or whatever their preferred means of communication.
- **Do they have any budget for this project?** Ideally try to use free solutions. However, sometimes you get what you pay for and there might be a great fit for a reasonable amount of money. Understand whether that is an option or not for your group.
- **What is their plan for supporting this project after the jam?** Who is going to be maintaining the content or process you are going to work on for the next 40+ hours? Understanding this plan can help you create good handover training and maintenance documentation (a great task for a non-technical teammate).

## Understanding the Who

> People are the heart of every nonprofit: staff, donors, volunteers, and the community they are serving. Each roles will likely be impacted by your project differently or use it in different ways. Ask your nonprofits who the primary targets of this solution are. If they say everyone, try to get them to rank which groups are the most important audiences.

Author: ```suggestion
People are the heart of every nonprofit: staff, donors, volunteers, and the community they are serving. Each role will likely be impacted by your project differently or use it in different ways. Ask your nonprofits who the primary targets of this solution are. If they say everyone, try to get them to rank which groups are the most important audiences.
```

![Journey Map][map]

## Form a Problem Statement
At this point you have enough information to form your problem statement. One way of framing this is in a question:

**How might we** [Insert Problem] (*optionally add:* **so that** [Desired Outcome])**?**

### Example from 2019 DSMHack
**Original solution idea from nonprofit:** Build a volunteer rewards store.

**Reframed problem statement:** How might we encourage volunteers to volunteer again after an initial organized volunteering event?

In the initial conversations the team discovered that there were also problems with their sign up forms that made it difficult to see upcoming events. The team delivered the asked for rewards store AND improved their volunteer sign up forms to better solve the problem from multiple angles.

## Prioritize
"If Our Work Together Could Only Achieve One Thing, What Would it Be?

## Additional Resources/Techniques

> https://www.designkit.org/methods

Author: ```suggestion
- https://www.designkit.org/methods
```
prefixing with `- ` makes a bulleted list

> https://www.atlassian.com/team-playbook/plays

Author: ```suggestion
- https://www.atlassian.com/team-playbook/plays
```
